@xen-orchestra/web/src/components/site/pools/PoolsTable.vue

```diff
@@ -0,0 +1,272 @@
+import type { XoServer } from '@/types/xo/server.type.ts'
+import VtsDataTable from '@core/components/data-table/VtsDataTable.vue'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import VtsStateHero from '@core/components/state-hero/VtsStateHero.vue'
+import UiButton from '@core/components/ui/button/UiButton.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiInfo, { type InfoAccent } from '@core/components/ui/info/UiInfo.vue'
+import UiLink from '@core/components/ui/link/UiLink.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTableActions from '@core/components/ui/table-actions/UiTableActions.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import { useRouteQuery } from '@core/composables/route-query.composable'
+import useMultiSelect from '@core/composables/table/multi-select.composable'
+import { useTable } from '@core/composables/table.composable'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import {
+  faCaretSquareDown,
+  faCity,
+  faEdit,
+  faEllipsis,
+  faEraser,
+  faHashtag,
+  faServer,
+} from '@fortawesome/free-solid-svg-icons'
+import { noop } from '@vueuse/shared'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { servers } = defineProps<{
+  servers: XoServer[]
+}>()
+
+const { t } = useI18n()
+
+const { isReady: isServerReady, hasError } = useServerStore().subscribe()
+const { isReady: isHostReady, get: getHostById } = useHostStore().subscribe()
+
+const selectedServerId = useRouteQuery('id')
+
+const searchQuery = ref('')
+
+const filteredServers = computed(() => {
+  const searchTerm = searchQuery.value.trim().toLocaleLowerCase()
+
+  if (!searchTerm) {
+    return servers
+  }
+
+  return servers.filter(server =>
+    Object.values(server).some(value => String(value).toLocaleLowerCase().includes(searchTerm))
+  )
+})
+
+const serverIds = computed(() => servers.map(server => server.id))
+
+const { selected, areAllSelected } = useMultiSelect(serverIds)
+
+const toggleSelect = () => {
+  selected.value = selected.value.length === 0 ? serverIds.value : []
+}
+
+const getStatus = (server: XoServer): { accent: InfoAccent; text: string } => {
+  if (server.error) {
+    return { accent: 'danger', text: t('error') }
+  }
+
+  switch (server.status) {
+    case 'disconnected':
+      return { accent: 'muted', text: t('disconnected') }
+    case 'connected':
+      return { accent: 'success', text: t('connected') }
+    case 'connecting':
+      return { accent: 'info', text: t('connecting') }
+    default:
+      return { accent: 'muted', text: t('unknown') }
+  }
+}
```

Comment:
What do you think of:
```suggestion
const getStatusInfo = createMapper(
  {
    error: { accent: 'danger', text: t('unable-to-connect-to-the-pool') },
    disconnected: { accent: 'muted', text: t('disconnected') },
    connected: { accent: 'success', text: t('connected') },
    connecting: { accent: 'info', text: t('connecting') },
  },
  'error'
)
```

And in the collection:
```ts
    define('status', record => getStatusInfo(record.error ? 'error' : record.status), { label: t('status') }),
```

cc @ByScripts